src/lints/trait_removed_associated_type.ron

```diff
@@ -0,0 +1,54 @@
+SemverQuery(
+    id: "trait_removed_associated_type",
+    human_readable_name: "associated type of trait removed",
+    description: "A trait's associated type was removed.",
+    required_update: Major,
+    reference_link: None,
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Trait {
+                        trait_name: name @output
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+
+                        importable_path {
+                            path @output @tag
+                        }
+
+                        associated_type {
+                            associated_type: name @output @tag
+
+                            span_: span {
+                                filename @output
+                                begin_line @output
+                            }
+                        }
+                    }
+                }
+            }
+            current @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+                item {
+                    ... on Trait {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+
+                        associated_type {
+                            name @filter(op: "=", value: ["%associated_type"])
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "zero": 0,
+    },
+    error_message: "A public trait's associated type was removed.",
```

Comment:
Use the "removed or renamed" wording to match other lints for removed items.

Reply:
[a5243a5](https://github.com/obi1kenobi/cargo-semver-checks/pull/500/commits/a5243a532875ac8315efe8769c803248e0ea5500)